src/immoney/registry.py

```diff
@@ -0,0 +1,19 @@
+from collections.abc import Mapping
+from typing import TypeAlias
+
+from immutables import Map
+
+from ._base import Currency
+
+CurrencyRegistry: TypeAlias = Mapping[str, Currency]
+
+
+class CurrencyCollector:
```

Comment:
I've been mucking a bit (too) much with `__slots__` recently. So that'd be the only thing I'd have to suggest here

```suggestion
class CurrencyCollector:
    __slots__ = ("__collection",)
```

Reply:
Good call 👍